### ISSUE TITLE
Parameters do not accept multiple constraints

### ISSUE THREAD

**Issue by sandermarechal** [FriendsOfSymfony/FOSRestBundle#1748]
It appears that parameters such as QueryParam or RequestParam do not accept multiple constraints. This makes some type of validations very hard to do. For example, I have a `page` parameter. The minimum value is 1 but there is no maximum so I cannot use `Range` but should use `GreaterThanOrEqual`. But that constraint also accepts things like dates, so I need to combine it with a `Type` constraint. I.e. something like:

```php
/**
* @QueryParam(name="page", requirements={
*     @Type("integer"),
*     @GreateThanOrEqual(5)
* })
*/
public function someAction(ParamFetcher $paramFetcher)
{
    // ...
}
```

**Comment by GuilhemN** [comment]
Can't you use `@All`?

**Comment by sandermarechal** [comment]
Nope, `@All` requires that the value to be checked is an array, not a scalar. It will throw an `UnexpectedTypeException`.

**Comment by GuilhemN** [comment]
Indeed you're right.
Well I guess it would make sense as under the hood we have an array of constraints, not sure how to detect it though. Can you submit a PR?

**Comment by sandermarechal** [comment]
Sure. PR created.

**Comment by alterphp** [comment]
:+1:

**Comment by sandermarechal** [comment]
Implemented, see #1752
